applications/addons/controllers/class.addoncontroller.php

```diff
@@ -665,7 +670,7 @@ public function browse($FilterToType = '', $Sort = '', $VanillaVersion = '', $Pa
             $Checked = $Session->getPreference('Addons.FilterChecked');
         }
 
-        if (!array_key_exists($FilterToType, AddonModel::$TypesPlural)) {
+        if (!array_key_exists($FilterToType, AddonModel::$TypesPlural) && $FilterToType != 'plugins,applications') {
```

Comment:
There's a lot to feel icky about in my changes here because the structure is so bad I have to hack in every change, but I understand what you're saying. Updated it.